### DISCUSSION TITLE
On Cloud use

### DISCUSSION THREAD

**Discussion by casolutions2** [bgregs514/vscode-tm1#29] [Ideas]
Hi, and great job kicking this off, I think it is a great concept.
I have installed the extension and completed the settings, however I am using an On Cloud version of PA and the format of the URL is slightly different to the On Prem version.  
I'm not sure on your knowledge of the On Cloud version, but it uses a format of https://_**client**_.planning-analytics.ibmcloud.com/tm1/api/_**clientserver**_/api/v1 no port is specified.  I believe the code in _src/net/net.ts_ function **initTM1Config** would need to be adapted to change **g_Config.baseURL** (when on cloud - which would be a setting?) to exclude the **:httpPortNumber** and to append the **clientServer** (again a setting?).
I am not a seasoned TS/JS developer and have no experience writing vscode extensions otherwise I would gladly fork the repo and attempt this myself.
I am happy to test, however, if one of the developers/contributors is OK with making the changes.
Thanks again! 
Craig.

**Comment by bgregs514** [comment]
Hi Craig,

Thank you for your feedback and for highlighting this issue!  Based on my (limited) knowledge of how the api works on the Cloud deployments, I believe you are exactly right - the fix should be a simple append to the baseURL variable.  In the short term we could add a simple checkbox toggle to the settings, but long term I think we should pursue some form of auto-detection (I also want to do this with auto-detecting instances from an admin host).

There are, unfortunately, a few things that are keeping me from actively contributing to this project at the moment, but I hope that they will be resolved soon.  I apologize for the delay and inconvenience, but I hope to be able to begin pushing out updates again in the near future!

Issue Link: #30

Thanks,

Ben
